src/components/HomePage/SearchArea/useUrlSearchParams.js

```diff
@@ -0,0 +1,127 @@
+import { useCallback, useMemo } from "react";
+
+import qs from "qs";
+import { useLocation, useHistory } from "react-router-dom";
+
+import { throttle } from "lodash";
+
+/**
+ * Injects into the Redux 'setXXXX' functions the ability to change the browser URL to reflect the specified change.
+ */
+export default ({
+    setJobDuration,
+    setShowJobDurationSlider,
+    setJobType,
+    setFields,
+    setTechs,
+    setSearchValue,
+    resetAdvancedSearchFields,
+} = {}) => {
+
+    const location = useLocation();
+    const history = useHistory();
+
+    const queryParams = useMemo(() => qs.parse(location.search, {
+        ignoreQueryPrefix: true,
+    }), [location]);
+
+    // need to throttle down calling 'history.replace' because
+    // too many invocations might cause the browser to stop responding
+    // value of 350 derived from https://github.com/PoorBillyPilgrim/artcrawl-digital-exhibit/issues/20#issue-761626379
+    // according to this 1000/3 is the lower bound for the 'wait' value
+    const changeURLFilters = useCallback(throttle((location, queryParams, changes) => {
+        const newQueryParams = {
+            ...queryParams,
+            ...changes,
+        };
+
+        history.replace({
+            ...location, search: qs.stringify(newQueryParams, {
+                skipNulls: true,
+                arrayFormat: "brackets",
+            }),
+        });
+    }, 350), [history]);
+
+    const clearURLFilters = useCallback(
+        throttle(
+            (location) => history.replace({ ...location, search: "" }),
+            350
+        ), [history]);
+
+    // destructure input here so mapDispatchToProps.setJobType is 'normalized' with respect to the other dispatch functions
```

Comment:
I didn't get it, what do you mean?

Reply:
In the `mapDispatchToProps` function, every function received the exact value that needed to be dispatched except for `setJobType` which received a DOM Event. I did this in order to have the function definitions "coherent" with eachother.